src/components/EnterpriseApp/data/enterpriseCurationReducer.js

```diff
@@ -54,12 +71,60 @@ function getHighlightSetsFromState(state) {
   return state.enterpriseCuration?.highlightSets || [];
 }
 
+/**
+ * Helper function to determine if there is a new archived content in a highlight set
+ *
+ * @param {Array} payload Array of highlight set objects
+ * [{
+ *  cardImageUrl: bool,
+ *  enterpriseCuration: string,
+ *  highlightedContent: array,
+ *  isPublished: bool,
+ *  title: string,
+ *  uuid: string,
+ * }]
+ * @returns {Boolean}
+ */
+function getIsNewArchivedContentFromState(payload) {
+  let isNewArchivedContent = false;
+  payload?.forEach(highlightSet => {
+    const dismissedCookies = global.localStorage.getItem(`${NEW_ARCHIVED_CONTENT_ALERT_DISMISSED_COOKIE_NAME}-${highlightSet.uuid}`);
+    // Checks that the highlightSet uuid isn't set
+    if (!dismissedCookies) {
+      highlightSet.highlightedContent.forEach(content => {
+        // If the length > 1 for the course run status e.g. ["archived", "published"]
+        // the content is not considered archived. An "unpublished" course run is not considered "archived" because
+        // it could have a scheduled course run in the future.
+        if (content.courseRunStatuses?.length === 1
+          && content.courseRunStatuses?.includes(COURSE_RUN_STATUSES.archived)
+        ) {
+          isNewArchivedContent = true;
+        }
+      });
+      // If the highlightSet uuid is already in the cookies, check if it includes the archived course content key
+    } else {
+      highlightSet.highlightedContent.forEach(content => {
+        if (
+          content.courseRunStatuses?.length === 1
+          && content.courseRunStatuses?.includes(COURSE_RUN_STATUSES.archived)
+          && !dismissedCookies.includes(content.contentKey)
+        ) {
+          isNewArchivedContent = true;
+        }
+      });
+    }
+  });
+  return isNewArchivedContent;
```

Comment:
nit: We could condense this logic
```suggestion
    const hasNewArchivedContent = (content) => {
      return content.courseRunStatuses?.length === 1
          && content.courseRunStatuses?.includes(COURSE_RUN_STATUSES.archived)
          && (!dismissedCookies || !dismissedCookies.includes(content.contentKey));
    };
  return highlightSet.highlightedContent.some(hasNewArchivedContent);
```

Reply:
thanks for this suggestion. I did some refactoring of this code and for readability I kept the `if-else` statement